ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -88,19 +99,27 @@ def add_experiences(
             if global_id in self.last_step_result:  # Don't store if agent just reset
                 self.last_take_action_outputs[global_id] = take_action_outputs
 
-        # Iterate over all the terminal steps
+        # Iterate over all the terminal steps, first gather all the teammate obs
+        # and then create the AgentExperiences/Trajectories
+        for terminal_step in terminal_steps.values():
+            self._gather_group_obs(terminal_step, worker_id)
```

Comment:
Will this only grab teammate obs that are also terminal? Should this be also grabbing from decision steps?